src/main/java/it/gov/pagopa/reward/service/commands/ops/DeleteInitiativeServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+                })
+                .then(rewardContextHolderService.refreshKieContainerCacheMiss())
+                .then();
+    }
+
+    private Mono<Void> deleteHpanInitiatives(String initiativeId){
+        return hpanInitiativesRepository.findAndRemoveInitiativeOnHpan(initiativeId)
+                .doOnNext(updateResult -> {
+                    log.info("[DELETE_INITIATIVE] Deleted {} instruments on initiative {}", updateResult.getModifiedCount(), initiativeId);
+                    auditUtilities.logDeletedHpanInitiative(initiativeId, updateResult.getModifiedCount());
+                })
+                .then();
+    }
+
+    private Mono<Void> deleteTransactionProcessed(String initiativeId){
+        return rewardContextHolderService.getInitiativeConfig(initiativeId)
+                .flatMap(initiativeConfig -> {
+                    if(InitiativeRewardType.DISCOUNT.equals(initiativeConfig.getInitiativeRewardType())){
+                        return transactionProcessedRepository.deleteByInitiativeId(initiativeId)
+                                .count()
+                                .doOnNext(count -> {
+                                    log.info("[DELETE_INITIATIVE] Deleted {} transactions on initiative {}", count, initiativeId);
+                                    auditUtilities.logDeletedTransaction(initiativeId, count);
+                                })
+                                .then();
+                    }
+                    else {
+                        return transactionProcessedRepository.findAndRemoveInitiativeOnTransaction(initiativeId)
+                                .doOnNext(updateResult -> {
+                                    log.info("[DELETE_INITIATIVE] Deleted {} transactions on initiative {}", updateResult.getModifiedCount(), initiativeId);
+                                    auditUtilities.logDeletedTransaction(initiativeId, updateResult.getModifiedCount());
+                                })
+                                .then();
+                    }
+                });
+    }
+
+    private Mono<Void> deleteEntityCounters(String initiativeId){
+        return userInitiativeCountersRepository.deleteByInitiativeId(initiativeId)
+                .map(UserInitiativeCounters::getUserId)
+                .doOnNext(entityId ->
+                        auditUtilities.logDeletedEntityCounters(initiativeId, entityId))
+                .then();
+    }
+
+
+    @Scheduled(cron = "${app.delete.initiative.clean}")
+    void scheduleRemovedAfterInitiativeDeletion(){
+        log.debug("[DELETE_INITIATIVE][SCHEDULE] Starting schedule to remove payment instruments and transactions after delete of initiatives");
+        removedAfterInitiativeDeletion()
+                .subscribe(x -> log.debug("[DELETE_INITIATIVE][SCHEDULE] Completed schedule to remove payment instruments and transactions after delete of initiatives"));
+    }
```

Comment:
why are you handling this as a @Scheduled?
just do it in the same reactive chain using then!

You are introducing a batch that will ever put in execution, performing queries evenif not necessary!